### ISSUE TITLE
GetApplicationDirectory() code is not up-to-date with modern FreeBSD

### ISSUE THREAD

**Issue by danfe** [enzo1982/smooth#5]
Currently, `GetApplicationDirectory()` works by calling `procstat -b <pid>` and getting the third field of the line matching the `<pid>`.  The problem is that modern FreeBSD versions return four fields:
```
% procstat -b $$
  PID COMM                OSREL PATH
42096 tcsh              1300132 /bin/tcsh
```
So `print $3` should really be `print $4`.  Another thing is that filtering the PID line could be avoided if `procstat` is called with `-h` (suppress header) switch.

Ideally, however, is not to spawn and chain any processes and read their output with `popen()`, but ask the system with `sysctl()` call for the `KERN_PROC_PATHNAME` as described [here](https://stackoverflow.com/questions/799679/programmatically-retrieving-the-absolute-path-of-an-os-x-command-line-app).

**Comment by enzo1982** [comment]
Thank you for bringing this up!

This is fixed in 7e21ca30 by changing to the `KERN_PROC_PATHNAME` sysctl. Also improved finding the binary path on macOS in the same commit.

**Comment by danfe** [comment]
Fixed in version 0.9.8, closing.
